### PR TITLE
feat: add `RemoteService`

### PR DESCRIPTION
**Pull request by crepererum**
Ref #2980.

### CODE REVIEW

File: generated_types/protos/influxdata/iox/management/v1/service.proto [hunk 1]
```diff
@@ -44,13 +44,19 @@ service ManagementService {
   rpc ListChunks(ListChunksRequest) returns (ListChunksResponse);
 
   // List remote IOx servers we know about.
-  rpc ListRemotes(ListRemotesRequest) returns (ListRemotesResponse);
+  rpc ListRemotes(ListRemotesRequest) returns (ListRemotesResponse) {
+    option deprecated = true;
```
Comment (alamb) [inline generated_types/protos/influxdata/iox/management/v1/service.proto:48]: Is the idea that once conductor is updated to use the new APIs, we can remove these deprecated ones?

If so, I suggest leaving a comment (with ticket reference) here with the backstory for the archaeological purposes of future readers

cc @domodwyer

File: influxdb_iox/src/influxdb_ioxd/server_type/database/rpc/mod.rs [hunk 2]
```diff
@@ -51,6 +52,12 @@ where
         operations::make_server(Arc::clone(server_type.application.job_registry()))
     );
 
+    // remotes can be managed even if the server is not serving
```
Comment (alamb) [inline influxdb_iox/src/influxdb_ioxd/server_type/database/rpc/mod.rs:55]: is the eventual plan to only add the relevant services for a particular `server_type`? (Aka only make a management server for `database` and only make remove service for `router`?

**Review by alamb — APPROVED** [review]
Looks good to me -- as I understand it this is effectively a copy/paste of the existing part of the management API (and a subsequent PR will remove the portions from the management API)

File: influxdb_iox/src/influxdb_ioxd/server_type/database/rpc/mod.rs (hunk 2 above)
Comment (crepererum) [reply influxdb_iox/src/influxdb_ioxd/server_type/database/rpc/mod.rs:55]: yes

File: generated_types/protos/influxdata/iox/management/v1/service.proto (hunk 1 above)
Comment (crepererum) [reply generated_types/protos/influxdata/iox/management/v1/service.proto:50]: yes, done

**Pull request merged**